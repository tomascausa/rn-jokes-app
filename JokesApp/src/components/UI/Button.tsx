import React from 'react';
import {Text, View, StyleSheet, Pressable} from 'react-native';
import {GlobalStyles} from '../../constants/styles';

interface ButtonProps {
    children: any;
    mode?: string;
    onPress?: () => void;
}

const Button = ({children, mode, onPress}: ButtonProps) => {
    return (
        <View>
            <Pressable
                onPress={onPress}
                style={({pressed}) =>
                    pressed && mode !== 'link' && styles.pressed
                }>
                <View style={mode === 'link' ? styles.link : styles.button}>
                    <Text style={styles.buttonText}>{children}</Text>
                </View>
            </Pressable>
        </View>
    );
};

const styles = StyleSheet.create({
    pressed: {
        opacity: 0.75,
        backgroundColor: GlobalStyles.colors.black,
        borderRadius: 4,
    },
    link: {},
    button: {
        padding: 15,
        borderRadius: 8,
        shadowColor: GlobalStyles.colors.black,
        shadowOffset: {
            width: 0,
            height: 3,
        },
        shadowOpacity: 0.3,
        shadowRadius: 8,
        elevation: 5,
        backgroundColor: GlobalStyles.colors.black,
    },
    buttonText: {
        fontFamily: 'Roboto-Regular',
        fontSize: 18,
        textAlign: 'center',
        color: GlobalStyles.colors.white,
    },
});

export default Button;
