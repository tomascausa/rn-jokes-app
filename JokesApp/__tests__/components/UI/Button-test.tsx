import 'react-native';
import React from 'react';
import Button from '../../../src/components/UI/Button';
import renderer from 'react-test-renderer';

describe('<Button />', () => {
    it('renders correctly', () => {
        const json = renderer.create(<Button>Test</Button>).toJSON();
        expect(json).toMatchSnapshot();
    });
});
