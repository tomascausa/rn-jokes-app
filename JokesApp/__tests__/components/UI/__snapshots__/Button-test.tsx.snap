// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
<View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={false}
  >
    <View
      style={
        Object {
          "backgroundColor": "#000000",
          "borderRadius": 8,
          "elevation": 5,
          "padding": 15,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 3,
            "width": 0,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Roboto-Regular",
            "fontSize": 18,
            "textAlign": "center",
          }
        }
      >
        Test
      </Text>
    </View>
  </View>
</View>
`;
